includes/base/controls-stack.php

```diff
@@ -444,7 +431,21 @@ public function add_control( $id, array $args, $options = [] ) {
 			}
 		}
 
-		if ( $should_optimize_controls ) {
+		if ( $this->should_optimize_controls() ) {
+			$ui_controls = [
+				Controls_Manager::RAW_HTML,
+				Controls_Manager::DIVIDER,
+				Controls_Manager::HEADING,
+				Controls_Manager::BUTTON,
+			];
+
+			if ( ! empty( $args['type'] ) && ! empty( $args['section'] ) && in_array( $args['type'], $ui_controls ) ) {
+				$args = [
+					'type' => $args['type'],
+					'section' => $args['section'],
+				];
+			}
```

Comment:
maybe now you can remove the unset in line 449?

Reply:
It's a different.